sstables/sstables.hh

```diff
@@ -101,6 +101,7 @@ extern size_t summary_byte_cost(double summary_ratio);
 struct sstable_writer_config {
     size_t promoted_index_block_size;
     size_t promoted_index_auto_scale_threshold;
+    bool force_promoted_index_for_first_partition = false;
```

Comment:
Why not make it unconditional and just always produce promoted index for first partition of all sstables?

Reply:
It will disturb small-partition loads a little. I agree the conditional is icky but introducing the promoted index code to small partition loads due to some random artifact is icky too.